relayer/src/transfer.rs

```diff
@@ -71,6 +75,57 @@ impl FromStr for Amount {
     }
 }
 
+#[derive(Copy, Clone)]
+pub enum TransferTimeout {
+    Height(Height),
+    Timestamp(Timestamp),
+    HeightAndTimestamp(Height, Timestamp),
+}
+
+impl TransferTimeout {
+    pub fn new(
+        timeout_height_offset: u64,
+        timeout: Duration,
+        destination_chain_status: &ChainStatus,
+    ) -> Result<Self, TransferError> {
```

Comment:
This is refactored from:

https://github.com/informalsystems/ibc-rs/blob/3e34b4b916eb91f680412e5fdd3ea045a99526d2/relayer/src/transfer.rs#L98-L111